app/equipment/weapon.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any
+
+from app.equipment.equipment import Equipment
+
+
+class Weapon(Equipment):
+    def __init__(self, name: str, power: int) -> None:
+        super().__init__(name)
+        self.power = power
+
+    @classmethod
+    def apply(cls, knight: Any, source: dict) -> None:
+        weapon_data = (
+            source.get(
+                knight.name.lower(),
+                source.get(
+                    knight.name.replace(" ", "_").lower(), {}
+                )
+            ).get("weapon", None)
+        )
+        if weapon_data:
+            knight.weapon = (

```

Comment:
weapon data is required
![image](https://github.com/user-attachments/assets/c22e99cc-480e-41a3-8519-a038b2431cde)